cmd/lakectl/cmd/root.go

```diff
@@ -153,9 +156,10 @@ const (
 	parallelismFlagName   = "parallelism"
 	noProgressBarFlagName = "no-progress"
 
-	defaultSyncParallelism = 25
-	defaultSyncPresign     = true
-	defaultNoProgress      = false
+	defaultParallelismConfig = 25
+	defaultSyncParallelism   = 0
```

Comment:
I'm not sure why both are needed.
We should:
1. Use configuration param if provided
2. Use modified param value if passed as part of command
3. Fallback to default param value

Reply:
My thought was, that positive default value should come from one source only. In case both have positive default values how will we determine which one to use? I chose the config val to be positive by default, following Ariel's [change request](https://github.com/treeverse/lakeFS/pull/8283#discussion_r1798350158) to go from the more general case to specific case.
In any case, this code run according to the priorities:
- no flag no config val - uses 25 as config default 
- no flag with config - uses config
- flag with no config - uses the flag, 
-  flag with config - uses flag

Reply:
Your priorities seem correct. We have a way to determine whether a flag in Cobra was explicitly provided by the user or not (there's at least one place in the code where we use it AFAIR).
You can use the same method to do that and determine whether we need to use the config value or the param value